tools/ci-scripts/linux/build.zsh

```diff
@@ -7,35 +7,39 @@ FLAVOR=$2
 BOOST=${3:-}
 COMPILER=${4:-clang}
 
-# We set our compilers and cflags in non-standard variables because Stack chokes
-# on some otherwise acceptable configurations. (e.g., CC="ccache gcc")
+export BOND_ROOT=/root/bond
+export BUILD_ROOT=/root/build
+BUILD_SCRIPTS=$BOND_ROOT/tools/ci-scripts/linux
+
+# We set our cflags in non-standard variables because Stack chokes on some
+# otherwise acceptable configurations.
 case "$COMPILER" in
     clang)
-        export CXX_COMPILER=clang++
-        export CXX_FLAGS="-Qunused-arguments --system-header-prefix=boost/"
-        export CC_COMPILER=clang
-        export CC_FLAGS="-Qunused-arguments --system-header-prefix=boost/"
+        export CXX=clang++
+        export CC=clang
+        export BOND_CXX_FLAGS="-Qunused-arguments --system-header-prefix=boost/"
+        export BOND_CC_FLAGS="-Qunused-arguments --system-header-prefix=boost/"
         ;;
 
     gcc)
-        export CXX_COMPILER=g++
-        export CXX_FLAGS=
-        export CC_COMPILER=gcc
-        export CC_FLAGS=
+        export CXX=g++
+        export CC=gcc
+        export BOND_CXX_FLAGS=
+        export BOND_CC_FLAGS=
         ;;
 
     *) echo "Unknown compiler $COMPILER"; exit 1;;
 esac
 
 export BOND_CMAKE_FLAGS="-DBOND_USE_CCACHE=TRUE"
+# Default boost root. C++ requires a specific boost and will override this.
+export BOOST_ROOT=/opt/boosts/boost_1_63_0
 
-BUILD_PATH=/root/build
-BUILD_SCRIPTS=/root/bond/tools/ci-scripts/linux
-
-mkdir -p $SYMLINKED_HOME $BUILD_PATH
+mkdir -p $SYMLINKED_HOME $BUILD_ROOT
+ln -s /root/.ccache $SYMLINKED_HOME/.ccache
 ln -s /root/.stack $SYMLINKED_HOME/.stack
 
-cd $BUILD_PATH
+cd $BUILD_ROOT
 
 case "$FLAVOR" in
     cpp-*)
```

Comment:
Why is this not moved to separate script as well?

Reply:
Just to reduce the complexity of the change. The only parameters to `hs`, `cs`, and `java` were already passed as envvars, meaning the scripts themselves require no arguments;`cpp-*` is a little more complex. I'm imagining the next change passing most or all of this script's argv to the flavor-specific script, but didn't want to tackle that in this PR.

Reply:
OK.